src/ext.rs

```diff
@@ -832,6 +858,31 @@ fn timeout2ms(dur: DWORD) -> Option<u32> {
 }
 
 #[cfg(any(feature = "nightly", feature = "duration"))]
+fn linger2dur(linger_opt: linger) -> Option<Duration> {
+    let dur = Duration::from_secs(linger_opt.l_linger as u64);
+    if dur.as_secs() == 0 {
+        None
+    }
+    else {
+        Some(dur)
+    }
+}
+
+#[cfg(any(feature = "nightly", feature = "duration"))]
+fn dur2linger(dur: Option<Duration>) -> linger {
+    match dur {
+        Some(d) => {
+            let onoff = if d.as_secs() == 0 { 0 } else { 1 };
+            linger {
+                l_onoff: onoff as u16,
```

Comment:
No, this is on purpose. According to [this](https://msdn.microsoft.com/de-de/library/windows/desktop/ms739165(v=vs.85).aspx) documentation *l_linger* is a nonzero value. So a duration with 0 seconds would cause a linger struct like `linger { l_onoff: 1, l_linger: 0 }` which is not ok.

The `None` part sets *l_linger* to zero, which is fine because l_onoff is set to 0.